### PR TITLE
[WINEMINE] Add Arabic (ar-SA and ar-EG) translation

### PR DESCRIPTION
**Pull request by WaltuhTabsDev**
## Purpose

Adding arabic support to winemine

JIRA issue: [CORE- 19496](https://jira.reactos.org/browse/CORE-19496)
## Proposed changes

Add Arabic to Winemine

### CODE REVIEW

**Comment by JoachimHenze** [comment]
Is arabic an right-to-left-language @Tabby1527 ?
Ftr: That would be our first ever arabic rc within ros. So if we want to accept it, we should make sure that the file-name (I18N-codename is fine).

File: base/applications/games/winmine/lang/ar-AR.rc
Comment (JoachimHenze) [inline base/applications/games/winmine/lang/ar-AR.rc:3]: license should most likely be LGPL. GPL is outdated in most of our files, and was used as license just in historic files. I do assume you copy pasted this from some existing file, right?

File: base/applications/games/winmine/lang/ar-AR.rc
Comment (JoachimHenze) [inline base/applications/games/winmine/lang/ar-AR.rc:4]: We do not mention the filename within headers anymore. That style is deprecated.

File: base/applications/games/winmine/lang/ar-AR.rc
Comment (JoachimHenze) [inline base/applications/games/winmine/lang/ar-AR.rc:8]: Also deprecated style. Current template for header files can be found here: https://reactos.org/wiki/Coding_Style#File_Structure

File: base/applications/games/winmine/lang/ar-AR.rc
Comment (binarymaster) [inline base/applications/games/winmine/lang/ar-AR.rc:10]:
```suggestion
/*
 * PROJECT:     ReactOS WineMine
 * LICENSE:     LGPL-2.1-or-later (https://spdx.org/licenses/LGPL-2.1-or-later)
 * PURPOSE:     Arabic resource file
 * TRANSLATOR:  Copyright 2024 Mahmoud Ahmed <your@email.com>
 */

```

File: base/applications/games/winmine/rsrc.rc
Comment (binarymaster) [inline base/applications/games/winmine/rsrc.rc:56]: There's no ar-AR, check existing Arabic locales here: https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/cmake/localization.cmake;hb=17b0623cf8428fa547e2c81d088bac1e0ceb96d1#l9

**Comment by binarymaster** [comment]
> That would be our first ever arabic rc within ros

Nope: https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/osk/lang/ar-DZ.rc;h=458a7dd2645b8a63cbece8e798f7ae81b57f7017;hb=17b0623cf8428fa547e2c81d088bac1e0ceb96d1

**Comment by Tabby1527** [comment]
> Is arabic a right-to-left-language @Tabby1527 ? Ftr: That would be our first ever arabic rc within ros. So if we want to accept it, we should make sure that the file-name (I18N-codename) is fine. Also of interest: Do we have all possible arabic glyphs within our Tahoma font? Would it require any additional font substitutions of "MS Shell Dlg"? Maybe @katahiromz can answer those questions.

Arabic (like Hebrew) is written from right to left , so yea

File: base/applications/games/winmine/rsrc.rc
Comment (Tabby1527) [reply base/applications/games/winmine/rsrc.rc:56]: There is no ar-001 , So I might change it to ar-sa instead

File: base/applications/games/winmine/lang/ar-AR.rc
Comment (Tabby1527) [reply base/applications/games/winmine/lang/ar-AR.rc:10]: alright

File: base/applications/games/winmine/lang/ar-EG.rc
Comment (binarymaster) [inline base/applications/games/winmine/lang/ar-EG.rc:4]:
```suggestion
 * PURPOSE:     Arabic (Egypt) resource file
```

File: base/applications/games/winmine/lang/ar-SA.rc
Comment (binarymaster) [inline base/applications/games/winmine/lang/ar-SA.rc:4]:
```suggestion
 * PURPOSE:     Arabic (Saudi Arabia) resource file
```

File: base/applications/games/winmine/lang/ar-EG.rc
Comment (binarymaster) [inline base/applications/games/winmine/lang/ar-EG.rc:8]:
```suggestion
LANGUAGE LANG_ARABIC, SUBLANG_ARABIC_EGYPT
```

File: base/applications/games/winmine/lang/ar-EG.rc
Comment (binarymaster) [inline base/applications/games/winmine/lang/ar-EG.rc:5]: E-mail specified here is supposed to be a real one and yours.

File: base/applications/games/winmine/lang/ar-SA.rc
Comment (binarymaster) [inline base/applications/games/winmine/lang/ar-SA.rc:8]:
```suggestion
LANGUAGE LANG_ARABIC, SUBLANG_ARABIC_SAUDI_ARABIA
```

**Review by binarymaster — COMMENTED** [review]
Just a few tweaks, please update your GitHub account settings, so maintainers can use Squash & Merge button without getting the author field modified back to nickname by GitHub, it's done in two steps:

1. Set your full name `Mahmoud Ahmed` in [GitHub profile settings](https://github.com/settings/profile) (the "Name" field there)
1. Uncheck `Keep my email addresses private` in [GitHub email settings](https://github.com/settings/emails) to avoid merging with noreply email

**Comment by WaltuhTabsDev** [comment]
> Just a few tweaks, please update your GitHub account settings, so maintainers can use Squash & Merge button without getting the author field modified back to nickname by GitHub, it's done in two steps:
> 
>     1. Set your full name `Mahmoud Ahmed` in [GitHub profile settings](https://github.com/settings/profile) (the "Name" field there)
> 
>     2. Uncheck `Keep my email addresses private` in [GitHub email settings](https://github.com/settings/emails) to avoid merging with noreply email

Done

**Comment by katahiromz** [comment]
Our `tahoma.ttf` and `tahomabd.ttf` have Arabic glyphs:



Arabic ReactOS is not installable due to locale problem:

```txt
(ntoskrnl/ex/locale.c:218) Failed to query value from Language Groups key (Status 0xc0000034)
err:(base/system/winlogon/winlogon.c:578) WL: Failed to initialize SAS
Terminating critical process 0xB07A65C8 (winlogon.exe)
Break, or Ignore (bi)? 
kdb:> 
```

**Review by katahiromz — APPROVED** [review]

**Comment by katahiromz** [comment]
Arabic ReactOS uses `UnicodeFonts` in `muifonts.h`.
I suggest use of `MS Shell Dlg` font substitute to `"Tahoma"` font.

**Review by JoachimHenze — CHANGES_REQUESTED** [review]
Thank you @katahiromz for your thoughts. I do recommend that we fix regression https://jira.reactos.org/browse/CORE-19496 **first before** we will add more arabic resources by merging this PR (although it's no hard blocker for the PR when being pedantic). The risk is low for this PR running into any merge-conflicts meanwhile, as it creates new files mostly. I do use the RequestChanges-button to indicate in the upper section of the PR that we need to wait a bit, although that is a bit stretched, as there is nothing wrong with the current state of this PR.

**Pull request closed without merging**
